internal/mux/frame.go

```diff
@@ -106,10 +106,26 @@ func (cs connSettings) maxPayloadSize() int {
 
 const connSettingsSize = 24
 
-var defaultConnSettings = connSettings{
-	RequestedPacketSize: 1440, // IPv6 MTU
-	MaxFrameSizePackets: 10,
-	MaxTimeout:          20 * time.Minute,
+func defaultConnSettings(conn net.Conn) (connSettings, error) {
```

Comment:
This change fixes a deadlock due to the host expecting to read 1460 bytes but renterd only sent 1440 bytes.

Reply:
lol yeah this is because the siamux handshake negotiates frame size by sending its desired settings... in a frame. 🤦🏻  Fixed in v2.